hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JsonKafkaSource.java

```diff
@@ -52,13 +61,30 @@ public JsonKafkaSource(TypedProperties properties, JavaSparkContext sparkContext
 
   @Override
   JavaRDD<String> toRDD(OffsetRange[] offsetRanges) {
-    JavaRDD<String> jsonStringRDD = KafkaUtils.createRDD(sparkContext,
+    JavaRDD<ConsumerRecord<Object, Object>> kafkaRDD = KafkaUtils.createRDD(sparkContext,
             offsetGen.getKafkaParams(),
             offsetRanges,
             LocationStrategies.PreferConsistent())
-        .filter(x -> !StringUtils.isNullOrEmpty((String) x.value()))
-        .map(x -> x.value().toString());
-    return postProcess(jsonStringRDD);
+        .filter(x -> !StringUtils.isNullOrEmpty((String) x.value()));
+
+    JavaRDD<String> result = shouldAppendKafkaOffsets() ? kafkaRDD.mapPartitions(partitionIterator -> {
```

Comment:
can we move the appending of offsets to a private method and keep this method lean (as you did for AvroKafkaSource)